packages/node-core/src/indexer/entities/Metadata.entity.ts

```diff
@@ -14,9 +37,39 @@ export type MetadataRepo = typeof Model & {
   new (values?: unknown, options?: BuildOptions): MetadataModel;
 };
 
-export function MetadataFactory(sequelize: Sequelize, schema: string): MetadataRepo {
+async function checkSchemaMetadata(sequelize: Sequelize, schema: string, chainId: string): Promise<boolean> {
+  try {
+    const res = await sequelize.query<Metadata>(`SELECT * FROM "${schema}"._metadata WHERE key = 'genesisHash'`, {
+      type: QueryTypes.SELECT,
+    });
+    return res[0]?.value === chainId;
+  } catch (e) {
+    return false;
+  }
+}
+
+export async function MetadataFactory(
+  sequelize: Sequelize,
+  schema: string,
+  multichain: boolean,
+  chainId: string
+): Promise<MetadataRepo> {
+  let tableName = '_metadata';
+
+  if (multichain) {
+    const oldMetadataName = await checkSchemaMetadata(sequelize, schema, chainId);
+
+    if (oldMetadataName) {
+      throw new Error(
```

Comment:
I think the message here be can be improved. 

It is like if we had a oldMetadata table here, we can not process any further. We need to tell them multi-chain indexing not compatible with old project, user should have `multi-chain` flag on when first time indexing . 

Either we are able to covert the old metadata table in future pr.